dataprofiler/data_readers/csv_data.py

```diff
@@ -148,26 +149,28 @@ def _guess_header_row(data_as_str, suggested_delimiter=None,
                     cell_type = 'none'
                 else:
             
-                    if cell.isdigit():
-                        cell_type = 'int'
-                        
-                    try:
-                        float(cell)
-                        cell_type = 'float'
-                    except ValueError:
-                        pass
-
                     try: 
-                        if dateutil.parser.parse(cell):
+                        if dateutil.parser.parse(cell, fuzzy=False):
                             cell_type = 'date'
                     except ValueError:
                         pass
                     except OverflowError:
                         pass
+
+                    try:
+                        float(cell)
+                        cell_type = 'float'
+                    except ValueError:
+                        pass
+
+                    if cell.isdigit():
+                        cell_type = 'int'
                     
                     if cell.isupper():
                         cell_type = 'upstr'
 
+                if cell_type not in ['str', 'none']:
+                    only_string_flag = False
```

Comment:
Checks if a document contains only strings. If so the standard functionality wont work